notes.js

```diff
@@ -4,17 +4,68 @@ app.use(express.json());
 const db = require('./db/connection');
 
 
-const { serverError, handleCustomErrors } = require('./errors');
+const { serverError, handleCustomErrors, psqlError } = require('./errors');
+const { fetchArticleId } = require('./models/GET-models');
 
 
+const getArticles = ((req, res, next) => {
+
+    console.log(req.query, "controller");
+    const { sortby, order, topic } = req.query;
+    console.log(req.query.hasOwnProperty('comment_count'));
+
+    const comment_count = req.query.hasOwnProperty('comment_count') ? 1 : 0;
+
+    fetchArticles(comment_count, sortby, order, topic).then((articles) => {
+
+        res.status(200).send({ articles: articles });
+
+    })
+        .catch((err) => {
+            console.log(err, "controller err");
+            next(err);
+        })
+
+})
+
+const fetchArticles = (comment_count, sortby = 'articles.created_at', order = 'DESC', topic = 'cats') => {
+
+    console.log(comment_count, "models");
+
+    const validCommentCount = [0, 1];
+    if (!validCommentCount.includes(comment_count)) {
+        return Promise.reject({ status: 400, msg: "Bad request" });
+    }
+
+    let str = `SELECT * FROM articles ORDER BY articles.created_at DESC`;
+
+    return db.query(str).then(({ rows }) => {
+
+        console.log(rows);
+        if (comment_count === 0) return rows;
```

Comment:
It would also help simplify this logic here as you won't need to check - you can just get the comment_count every time